### ISSUE TITLE
Extremely slow training convergence speed and low performance compared to the original darknet

### ISSUE THREAD

**Issue by nonstop1962** [thtrieu/darkflow#904]
I try hard to parse my custom project in the original darknet framework into the tensorflow framework using darkflow code. 

Although I have tried my best to configure the same training settings as possible for the darkflow, darkflow frame shows significantly slower training convergence speed than the darknet framework.

My custom project has 6 classes and each class has 130 training images. 
In the original darknet framework, the loss value at the beginning is around 15 and converges around 1500 iterations with the loss value of 0.1.
On the other hand, the darkflow framework shows an initial loss value of 100 and converge after more than 20,000 iterations with the loss value of 1.0.
Also, the detection performance of the darknet framework is much better than the darkflow (darknet can catch all the object in the test images but darkflow cannot).

I'm analyzing the source code of the darkflow and checking the following items to make the same training settings as the darknet.
- training batch size: both frameworks using 'batch=16'
- optimizer: darknet - momentum=0.9, decay=0.0005 (in yolo-voc.2.0.cfg) vs darkflow - momentum optimizer with momentum=0.9
- learning rate policy: darknet - learning_rate=0.0001, policy=steps, steps=100,25000,35000 (in yolo-voc.2.0.cfg) vs darkflow - same setting using PR#216
- pre-trained weight: both frameworks using 'darknet19_448.conv.23'

I also checked the data augmentation method (rotation, flip, jittter, etc.) and data preprocessing (load image with opencv and divide its pixel value with 255.) in darkflow, but I think there are same with the darknet. So I'm now trying to investigate the loss function and loading pre-trained weights part of the darkflow code.
Does anyone have the exact same implementation of darknet's training phase in the darkflow framework? Can I get some hints to do this?

**Comment by nonstop1962** [comment]
+ I pretty make sure that preparation for my custom dataset is correctly done (label.txt, annotations, images).

**Comment by tomsang** [comment]
Thus what's the benefit to use darkflow?

**Comment by tomsang** [comment]
See https://www.linkedin.com/pulse/comparison-yolo-darknet-versus-darkflow-rasmus-kromann

I think the performance penalty of darkflow is due to design of tensorflow.
